src/azul/service/avro_pfb.py

```diff
@@ -423,22 +442,26 @@ def string_to_datetime(data, *args):
 
 
 def _entity_schema_recursive(field_types: FieldTypes,
-                             files_entity: bool = False) -> Iterable[JSON]:
+                             files_entity: bool = False,
+                             path: Tuple[str, ...] = ()) -> Iterable[JSON]:
     for entity_type, field_type in field_types.items():
         plural = isinstance(field_type, list)
         if plural:
             field_type = one(field_type)
         if isinstance(field_type, dict):
             yield {
                 "name": entity_type,
+                "namespace": '.'.join(path),
                 "type": {
                     # This is always an array, even if singleton is passed in
                     "type": "array",
                     "items": {
                         "name": entity_type,
+                        "namespace": '.'.join(path),
```

Comment:
What does this do? It seems unrelated.

Reply:
Also, shouldn't `'.'.join(path)` be extracted into a variable to avoid duplicating the expression?

Reply:
> What does this do? It seems unrelated.

https://github.com/DataBiosphere/azul/pull/3380#issuecomment-910465259

Reply:
Makes sense.